### PR TITLE
Fix #2278: Add RTL support for Admin Controls Reset Pin

### LINKED ISSUE #2278: [RTL] High-fi Admin Controls Reset Pin
`Admin Controls Reset Pin` needs RTL support as shown in below screen.
AC EP (Reset PIN, Empty) – RTL

Make sure that this works correctly in mobile (landscape + portrait) and tablet (landscape + portrait).

## References
Mobile Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/grid/
Mobile Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/grid/
Tablet: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/grid/
Wiki: https://github.com/oppia/oppia-android/wiki/Working-on-UI

### PR DESCRIPTION
**Pull request by ShridharGoel**
## Explanation
Fixes #2278

## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by FareesHussain** [comment]
@rt4914 as ProfileInputView is implemented in many layouts it would be better if we implement RTL within ProfileInputView to reduce lines of code in XML 

What do you think?

**Comment by FareesHussain** [comment]
> @rt4914 as ProfileInputView is implemented in many layouts it would be better if we implement RTL within ProfileInputView to reduce lines of code in XML
> 
> What do you think?

This is not required ProfileInputView already supports rtl

File: app/src/main/res/layout-land/profile_reset_pin_activity.xml
Comment (FareesHussain) [inline app/src/main/res/layout-land/profile_reset_pin_activity.xml]: This not required ProfileInputView already supports RTL

File: app/src/main/res/layout-land/profile_reset_pin_activity.xml
Comment (FareesHussain) [inline app/src/main/res/layout-land/profile_reset_pin_activity.xml]: ditto here

File: app/src/main/res/layout/profile_reset_pin_activity.xml
Comment (FareesHussain) [inline app/src/main/res/layout/profile_reset_pin_activity.xml]: ditto here

File: app/src/main/res/layout/profile_reset_pin_activity.xml
Comment (FareesHussain) [inline app/src/main/res/layout/profile_reset_pin_activity.xml]: ditto here

**Review by FareesHussain — CHANGES_REQUESTED** [review]
@ShridharGoel Please check

**Comment by ShridharGoel** [comment]
> This is not required ProfileInputView already supports rtl

@FareesHussain Where is this implemented? Support for RTL does not seem to be there in `ProfileInputView.kt`.

**Comment by FareesHussain** [comment]
> > This is not required ProfileInputView already supports rtl
> 
> @FareesHussain Where is this implemented? Support for RTL does not seem to be there in `ProfileInputView.kt`.

I mean RTL works for `ProfileInputView` without adding `android:textDirection="locale"`

**Comment by ShridharGoel** [comment]
> I mean RTL works for ProfileInputView without adding android:textDirection="locale"

The input text does not show in RTL format by default.

**Comment by ShridharGoel** [comment]
In the current changes of this PR as well, the input is not supporting RTL. I am updating it to work as expected.

**Comment by ShridharGoel** [comment]
Updated it so that input text supports RTL as well. Also, the change is in `ProfileInputView.kt` so it will be applicable everywhere `ProfileInputView` is used.

**Comment by FareesHussain** [comment]
@ShridharGoel Need reformating. I suggest to set up a macro so it will be handy all the time, and it also works with XML files. 
Check out wiki here - https://github.com/oppia/oppia-android/wiki/Ktlint-Guide#macros

**Comment by FareesHussain** [comment]
LGTM, 
Workflow needs to be re-run as it passed if run on local machine
image

image

**Comment by rt4914** [comment]
> LGTM,
> Workflow needs to be re-run as it passed if run on local machine
> image
> 
> image

@FareesHussain Thanks for reviewing this PR. Normally you should not write LGTM unless you have the author has made all the changes suggested by you and you have resolved all the comments that you started.

File: app/src/main/res/layout-land/profile_reset_pin_activity.xml
Comment (rt4914) [reply app/src/main/res/layout-land/profile_reset_pin_activity.xml]: @ShridharGoel Please reply to these comments.

File: app/src/main/res/layout-land/profile_reset_pin_activity.xml
Comment (rt4914) [reply app/src/main/res/layout-land/profile_reset_pin_activity.xml]: @ShridharGoel Please reply to these comments.

File: app/src/main/res/layout/profile_reset_pin_activity.xml
Comment (rt4914) [reply app/src/main/res/layout/profile_reset_pin_activity.xml]: @ShridharGoel Please reply to these comments.

File: app/src/main/res/layout/profile_reset_pin_activity.xml
Comment (rt4914) [reply app/src/main/res/layout/profile_reset_pin_activity.xml]: @ShridharGoel Please reply to these comments.

File: app/src/main/res/layout-land/profile_reset_pin_activity.xml
Comment (ShridharGoel) [reply app/src/main/res/layout-land/profile_reset_pin_activity.xml]: This has been updated.

File: app/src/main/res/layout-land/profile_reset_pin_activity.xml
Comment (ShridharGoel) [reply app/src/main/res/layout-land/profile_reset_pin_activity.xml]: This has been updated.

File: app/src/main/res/layout/profile_reset_pin_activity.xml
Comment (ShridharGoel) [reply app/src/main/res/layout/profile_reset_pin_activity.xml]: This has been updated.

File: app/src/main/res/layout/profile_reset_pin_activity.xml
Comment (ShridharGoel) [reply app/src/main/res/layout/profile_reset_pin_activity.xml]: This has been updated.

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

@ShridharGoel Please reply to all the comment threads mentioned in this PR.

@FareesHussain Once @ShridharGoel replies to all the comments that you have mentioned, you can resolve these comments.


For @BenHenning 's reference:

Screenshot_1608272580
Screenshot_1608272630

**Comment by ShridharGoel** [comment]
@rt4914 Replied to all the comments.

**Comment by rt4914** [comment]
> @rt4914 Replied to all the comments.

Thanks. @ShridharGoel You can pick next issue and start working on it. It will be reviewed by @BenHenning and merge after that.

@FareesHussain You can resolve the comments if the replies makes sense.

**Comment by BenHenning** [comment]
Thanks @ShridharGoel. I might be missing some context, but why does textDirection need to be set at all? Is this just to force RTL when using the developer setting? I'm assuming that if we actually were displaying strong RTL characters (such as Arabic) that RTL would work correctly with the existing views. Is that not the case?

**Comment by ShridharGoel** [comment]
@BenHenning It is required for the TextView. RTL is not working for the TextView without setting the `textDirection` as `locale`.

**Comment by BenHenning** [comment]
@ShridharGoel sorry, I may not have been clear enough in my earlier comment: have you tried to create a strings.xml for an RTL language (such as Arabic) with one of these strings set to an Arabic translation (e.g. by using Google Translate) & then opened the app with Arabic set as the system language? I would expect that per the Android defaults RTL _will_ work for these text views and that the reason it's appearing to not work is because we're not actually using an RTL language, we're just forcing RTL layout.

**Comment by BenHenning** [comment]
(note that I'm going to be largely unavailable for the next 2 weeks, so deferring to @rt4914 to make a decision on whether this is actually necessary, and whether we should be determining a way to make this change across the codebase since it's an extra property that seems to only affect developers).

Rajat: it may also be worth considering whether introducing an ar-XB locale may work better (ar-XB is reversed English letters with strong RTL characters to force RTL layouts).

**Comment by ShridharGoel** [comment]
@BenHenning Checked, the `textDirection` attribute is not required. Updated accordingly.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Blocked on #2330

**Comment by ShridharGoel** [comment]
@rt4914 Please have a look since #2330 is merged now.

**Comment by rt4914** [comment]
> Blocked on #2330

@ShridharGoel Apologies for not being clear in my last comment. When I wrote blocked it also meant that you will need to merge your branch with develop when that PR is merged.

Also, the outcome of your changes is 
Screenshot_1609218202
You can notice that the description is not working correctly.

To test this correctly merge your branch with latest `develop` and follow steps mentioned in this https://github.com/oppia/oppia-android/wiki/RTL-Guidelines and based on the result update your xml files.

Test app for `cibarA [XB]` language and fix UI as per your requirements.

Video Reference: https://www.youtube.com/watch?v=gc5HeuksMhc&list=PLoDt3UyLUtcgZLxUPcmyaJIajXfO62I7M&index=1

**Comment by ShridharGoel** [comment]
@rt4914 I have merged this branch with `develop`.

> You can notice that the description is not working correctly.

Any idea what can be the reason? I tried few changes but it did not seem to work. Same issue is there in the screenshot which is present at this link as well: https://github.com/oppia/oppia-android/wiki/RTL-Guidelines

**Review by FareesHussain — APPROVED** [review]

**Comment by rt4914** [comment]
> @rt4914 I have merged this branch with `develop`.
> 
> > You can notice that the description is not working correctly.
> 
> Any idea what can be the reason? I tried few changes but it did not seem to work. Same issue is there in the screenshot which is present at this link as well: https://github.com/oppia/oppia-android/wiki/RTL-Guidelines

@ShridharGoel Your PR is correct. Turns out that `pseudo-locale` does not work for multiple lines. I have found any such issue on internet but its the observation that I made going through different screens in our app. 
Also, I tested this by introducing the actual `Arabic` language string in the app and the result was correct.
Screenshot_1609308193

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**
